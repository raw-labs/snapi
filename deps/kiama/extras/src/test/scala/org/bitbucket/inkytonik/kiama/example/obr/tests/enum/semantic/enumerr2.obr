(* Enumeration-related errors in statements *)

PROGRAM EnumErr2 (i : INTEGER) : INTEGER;

VAR
    CarColour : (Red, Blue, Green);
    WeekDays: (Monday, Tuesday, Wednesday, Thursday, Friday);

BEGIN
    (* Type incompatibility between enum var and enum constant in
       assignment statement *)
    CarColour := Wednesday;

    (* Enumeration constant is not assignable *)
    Red := Blue;

    (* Type incompatibility between enum var and enum constant
       in expression *)
    IF Blue < Tuesday THEN
        i := 0;
    END

    (* Double check that we can't compare Boolean values with < *)
    IF FALSE < TRUE THEN
        i := 1;
    END

    (* Non enumeration constant used with enumeration variable *)
    CarColour := 0;

    (* Enumeration constant used with non enumeration variable *)
    i := Red;

    (* Enumeration value used with enumeration value of same type *)
    CarColour := Red + Green;

    (* Enumeration value used with enumeration value of other type *)
    CarColour := Red + Monday;

    (* Enumeration value used with non-enumeration value *)
    i := Red + 1;
END EnumErr2.
