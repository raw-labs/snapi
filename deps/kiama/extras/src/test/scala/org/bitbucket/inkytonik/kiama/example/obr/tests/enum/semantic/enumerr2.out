src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:12:18:error: expected enumeration CarColour type got enumeration WeekDays
    CarColour := Wednesday;
                 ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:15:5:error: illegal assignment
    Red := Blue;
    ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:19:15:error: expected enumeration CarColour type got enumeration WeekDays
    IF Blue < Tuesday THEN
              ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:24:8:error: expected integer or any enumeration type got boolean
    IF FALSE < TRUE THEN
       ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:29:18:error: expected enumeration CarColour type got integer
    CarColour := 0;
                 ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:32:10:error: expected integer type got enumeration CarColour
    i := Red;
         ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:35:18:error: expected integer type got enumeration CarColour
    CarColour := Red + Green;
                 ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:35:22:error: expected enumeration CarColour type got integer
    CarColour := Red + Green;
                     ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:35:24:error: expected integer type got enumeration CarColour
    CarColour := Red + Green;
                       ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:38:18:error: expected integer type got enumeration CarColour
    CarColour := Red + Monday;
                 ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:38:22:error: expected enumeration CarColour type got integer
    CarColour := Red + Monday;
                     ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:38:24:error: expected integer type got enumeration WeekDays
    CarColour := Red + Monday;
                       ^
src/test/scala/org/bitbucket/inkytonik/kiama/example/obr/tests/enum/semantic/enumerr2.obr:41:10:error: expected integer type got enumeration CarColour
    i := Red + 1;
         ^
